api/auth/index.ts

```diff
@@ -1,179 +1,163 @@
+    if (!app.locals.issuer) {
+        try {
+            app.locals.issuer = await configureIssuer(idamUrl)
+        } catch (error) {
+            return next(error)
+        }
+    }
 
-    const url = `${getConfigValue(SERVICES_IDAM_API_PATH)}/oauth2/token?grant_type=authorization_code&code=${req.query.code}&redirect_uri=${
-      getConfigValue(PROTOCOL)
-  }://${req.headers.host}${getConfigValue(OAUTH_CALLBACK_URL)}`
+    if (!app.locals.client) {
+        const clientMetadata: ClientMetadata = {
+            client_id: idamClient,
+            client_secret: secret,
+            response_types: ['code'],
+            token_endpoint_auth_method: 'client_secret_post', // The default is 'client_secret_basic'.
+        }
 
-    return http.post(url, {}, options)
-}
+        app.locals.client = new app.locals.issuer.Client(clientMetadata)
+    }
 
-async function sessionChainCheck(req: EnhancedRequest, res: express.Response, accessToken: string) {
-    if (!req.session.auth) {
-        logger.warn('Session expired. Trying to get user details again')
-        console.log(getUserDetails(accessToken, idamUrl))
-        const details = await asyncReturnOrError(getUserDetails(accessToken, idamUrl), 'Cannot get user details', res, logger, false)
-
-        if (details) {
-            logger.info('Setting session')
-
-            //const orgIdResponse = await getOrganisationId(details)
-            // no real end point
-            const orgIdResponse = {
-                data: {
-                    id: '1',
-                },
-            }
-            req.session.auth = {
-                email: details.data.email,
-                orgId: orgIdResponse.data.id,
-                roles: details.data.roles,
-                token: accessToken,
-                userId: details.data.id,
-            }
+    const host = req.get('host')
+    if (host.indexOf(':') > 0 ) {
+        const hostwithoutPort = host.substring(0, host.indexOf(':'))
+        if (net.isIP(hostwithoutPort)) {
+            return next()
         }
     }
 
-    if (!req.session.auth) {
-        logger.warn('Auth token  expired need to log in again')
-        doLogout(req, res, 401)
-        return false
-    }
+    console.log('host is', host)
+    const fqdn = getConfigValue(PROTOCOL) + '://' + host
+    const redirectUri = `${fqdn}/${getConfigValue(OAUTH_CALLBACK_URL)}`
+    console.log('redirectUri', redirectUri)
 
-    return true
-}
+    // logger.info('configuring strategy with redirect_uri:', redirectUri)
+
+    passport.unuse('oidc').use('oidc', new Strategy({
+        client: app.locals.client,
+        params: {
+            prompt: 'login',
+            redirect_uri: redirectUri,
+            scope: 'profile openid roles manage-user create-user',
+        },
+        sessionKey: 'xui_webapp', // being explicit here so we can set manually on logout
+        usePKCE: false, // issuer doesn't support pkce - no code_challenge_methods_supported
```

Comment:
Think about using block level comments over in-line commenting, something for later.